sdk/typescript/src/transactions/Transaction.ts

```diff
@@ -410,20 +410,24 @@ export class Transaction {
 
 	// Method shorthands:
 
-	splitCoins(
-		coin: TransactionObjectArgument | string,
-		amounts: (TransactionArgument | SerializedBcs<any> | number | string | bigint)[],
-	) {
-		return this.add(
-			Commands.SplitCoins(
-				typeof coin === 'string' ? this.object(coin) : this.#resolveArgument(coin),
-				amounts.map((amount) =>
-					typeof amount === 'number' || typeof amount === 'bigint' || typeof amount === 'string'
-						? this.pure.u64(amount)
-						: this.#normalizeTransactionArgument(amount),
-				),
+	splitCoins<
+		const Amounts extends (TransactionArgument | SerializedBcs<any> | number | string | bigint)[],
+	>(coin: TransactionObjectArgument | string, amounts: Amounts) {
+		const command = Commands.SplitCoins(
+			typeof coin === 'string' ? this.object(coin) : this.#resolveArgument(coin),
+			amounts.map((amount) =>
+				typeof amount === 'number' || typeof amount === 'bigint' || typeof amount === 'string'
+					? this.pure.u64(amount)
+					: this.#normalizeTransactionArgument(amount),
 			),
 		);
+		const index = this.#data.commands.push(command);
+		return createTransactionResult(index - 1, amounts.length) as Extract<
+			Argument,
+			{ Result: unknown }
```

Comment:
Curious why the unknown on the results and nested results here

Reply:
Just copied these types from other places.  Generally unknown works well with extract, the resulting type will preserve the more specific type without you having to know what it is